attribute/set.go

```diff
@@ -238,19 +228,23 @@ func NewSetWithFiltered(kvs []KeyValue, filter Filter) (Set, []KeyValue) {
 //
 // The second []KeyValue return value is a list of attributes that were
 // excluded by the Filter (if non-nil).
-func NewSetWithSortableFiltered(kvs []KeyValue, tmp *Sortable, filter Filter) (Set, []KeyValue) {
+func NewSetWithSortableFiltered(kvs []KeyValue, _ *Sortable, filter Filter) (Set, []KeyValue) {
 	// Check for empty set.
 	if len(kvs) == 0 {
 		return empty(), nil
 	}
 
-	*tmp = kvs
-
 	// Stable sort so the following de-duplication can implement
 	// last-value-wins semantics.
-	sort.Stable(tmp)
-
-	*tmp = nil
+	slices.SortStableFunc(kvs, func(a, b KeyValue) int {
+		if a.Key == b.Key {
+			return 0
+		}
+		if a.Key < b.Key {
+			return -1
+		}
+		return 1
```

Comment:
Use https://pkg.go.dev/cmp#Compare

```suggestion
		return cmp.Compare(a.Key, b.Key)
```